src/state.rs

```diff
@@ -550,6 +551,16 @@ impl Heap {
         result
     }
 
+    pub fn read_unaligned_from_pointer(&self, pointer: &FatPointer) -> Vec<u8> {
```

Comment:
Why doesn't this work with the normal read from pointer?

Reply:
Because this returns a `Vec<u8>` instead of a `U256`.

Reply:
You can create the `Vec<u8>` from the `U256` (or viceversa), I don't think it is a good a idea to have to functions that do almost the same, in any case, replace the implementation of one of them with the other plus a conversion